app/src/components/widgets/charts/cumulativeTrendChart/cumulativeTrendChart.jsx

```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from 'react-intl';
+import ReactDOMServer from 'react-dom/server';
+import classNames from 'classnames/bind';
+import { C3Chart } from 'components/widgets/charts/common/c3chart';
+import { messages as commonMessages } from '../common/messages';
+import { Legend } from '../common/legend';
+import styles from './cumulativeTrendChart.scss';
+import { CumulativeTrendTooltip } from './tooltip';
+import { generateChartDataParams, generateChartColors, getColorForKey } from './generateConfig';
+
+const cx = classNames.bind(styles);
+
+const messages = {
+  statistics$executions$failed: commonMessages.failed,
+  statistics$executions$skipped: commonMessages.skipped,
+  statistics$executions$passed: commonMessages.passed,
+  statistics$defects$product_bug$total: commonMessages.pb,
+  statistics$defects$automation_bug$total: commonMessages.ab,
+  statistics$defects$system_issue$total: commonMessages.si,
+  statistics$defects$no_defect$total: commonMessages.nd,
+  statistics$defects$to_investigate$total: commonMessages.ti,
+};
+
+@injectIntl
+export class CumulativeTrendChart extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool,
+    height: PropTypes.number,
+    container: PropTypes.instanceOf(Element).isRequired,
+    observer: PropTypes.object,
```

Comment:
Please provide props to manage widget legend and use it inside of your component.
See f.e. `widgets/charts/launchStatisticsChart/launchStatisticsChart.jsx` lines 63, 64.

Reply:
done